_drafts/intro-to-working-with-data-in-CUBA.adoc

```diff
@@ -0,0 +1,201 @@
+
+
+The part that loads the Pet instance by its ID looks like this with the use of the DataManager API:
+
+
+.VisitServiceBean.java
+[source,java]
+----
+@Inject
+DataManager dataManager;
+
+Pet loadPetById(String petId) {
+  return dataManager.load(Pet.class)
+            .query("select e from petclinic$Pet e where e.petId = :petId")
+            .parameter("petId", petId)
+            .one();
+}
+----
+
+DataManager has a https://martinfowler.com/bliki/FluentInterface.html[fluent API] which allows to chain operations on the load operation in order to define parameters to the execution. In this case, the interaction starts with `dataManager.load(Pet.class)`. Then the JPQL query is defined and the associated parameters are configured with the parameter values. The method call `one()` executes the load operation and fetches exactly one item that matches the criteria. If there are no results or multiple results the corresponding exceptions are thrown.
+
+It is up to the application developer to decide if the query can return one or possibly multiple results. In this case it is only possible to create a Pet ID once, we can be sure that the query will return at the most one item.
+
+In this case, the attribute `petId` is not the primary key of the entity. If this would be the case, the query would be even easier to create as the query part is not needed: `dataManager.load(Pet.class).id(petId).one();`
+
+=== Multiple Items by query
+
+The next very common option is to load multiple Items defined by a query.
+
+The Petclinic application has the capability to let the user send out Disease Warning Mailings. The user has to define the location of a known disease as well as the type of Pet that is endangered by this disease. The system will find all endangered Pets and send out a Mailing to the owners.
+
+image:https://dummyimage.com/600x400/00b5ec/ffffff.png&text=Disease Warning Mailing[] Disease Warning Mailing overview
+
+
+The part that finds the endangered Pets looks like this with the use of the DataManager API:
+
+.DiseaseWarningMailingServiceBean.java
+[source,java]
+----
+@Inject
+DataManager dataManager;
+
+private List<Pet> findEndangeredPets(PetType petType, String city) {
+    return dataManager.load(Pet.class)
+                .query("select e from petclinic$Pet e where e.owner.city = :ownerCity and e.type.id = :petType")
+                .parameter("ownerCity", city)
+                .parameter("petType", petType)
+                .view("pet-with-owner-and-type")
+                .list();
+}
+----
+
+The query part is a little more complicated compared to above, but the same pattern applies. The differences in this example are the usage of the `list()` method as we expect multiple Pets from the database query. Also a specific view has been used, because not only the Pet data, but also the Owner data has to be loaded in order to send out the Mailings.
+
+== Write data with DataManager
```

Comment:
Write Data with DataManager
plz make sure all headings are in Title Case

Reply:
I wasn't aware of that this is common in english. It this really the case?